stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -195,3 +219,32 @@ function eigvals!(A::Hermitian{T,S}, B::Hermitian{T,S}; sortby::Union{Function,N
     return vals
 end
 eigvecs(A::HermOrSym) = eigvecs(eigen(A))
+
+# Note: No specilized LAPACK routines for Matrix+Symmetric and Symmetric+Matrix exist. Hence, the calls are forwarded to conventional Matrix eigvals!
+function eigvals!(A::StridedMatrix{T}, B::Union{RealHermSymComplexHerm{T},Diagonal{T}}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    return eigvals!(A, Matrix{T}(B); sortby)
+end
+
+function eigvals!(A::RealHermSymComplexHerm{T,<:StridedMatrix}, B::AbstractMatrix{T}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    return eigvals!(Matrix{T}(A), B; sortby)
+end
```

Comment:
I believe @jishnub is talking about these two, and I agree. If we can't do it in-place, then that should give a `MethodError`. If users want to get the result by all means and don't care about memory, then they should use `eigvals`, or else make sure they provide feasible matrices.

Reply:
Hello @jishnub and @dkarrasch, thank you for the comments. I agree too and have simply removed the `!` from the function names. Attempting to call `eigen!` or `eigvals!` with a single symmetric/Hermitian parameter will result in a method error, as suggested. Nevertheless, with this change as well, issue #49533 is resolved.